src/coreclr/gc/gcinterface.h

```diff
@@ -952,11 +952,32 @@ inline GC_ALLOC_FLAGS& operator&=(GC_ALLOC_FLAGS& a, GC_ALLOC_FLAGS b)
 #define UNCHECKED_OBJECTREF_TO_OBJECTREF(obj)       (obj)
 #endif
 
+#if !defined(FEATURE_HIJACK) || defined(TARGET_UNIX)
+#define FEATURE_SUPPORTS_STACK_LIMIT
+#else
+#if defined(FEATURE_CONSERVATIVE_GC)
+// On Windows platforms when we hijack, we use SuspendThread from outside the thread, and if we suspend
+// within an interruptible region, we can stop with the top Frame that is an InlinedCallFrame
```

Comment:
I would expect `RedirectedThreadFrame` to be the top Frame when we stop within an interruptible region and start GC walk. Why are the problematic cases missing `RedirectedThreadFrame` ?

Reply:
You have a point. I thought I had managed to reproduce the scenario (or something very like it) in the debugger, but I agree, I can't quite see how we didn't end up with a proper RedirectedThreadFrame. In any case, the best crash dump I could get from the CI system had a disabled InlinedCallFrame as the top of the framestack. (Unfortunately, for some reason, the thread that frame is on had been terminated, so it wasn't completely workable.) I'll try a again with some debugger fun times.